client/app/components/routes/select-operator/form/UserOperator.tsx

```diff
@@ -53,38 +58,47 @@ const createUserOperatorSchema = (
         .pc_business_structure,
       anyOf: businessStructureOptions,
     };
+
   return localSchema;
 };
 
 export default async function UserOperator({
   params,
 }: Readonly<{
-  params?: { id?: number };
+  params?: { id?: number; readonly?: boolean };
 }>) {
-  const serverError = <div>Server Error. Please try again later.</div>;
-
   const businessStructures: BusinessStructure[] | { error: string } =
     await getBusinessStructures();
 
   const userData: UserInformationInitialFormData | { error: string } =
     await getCurrentUser();
 
-  if ("error" in userData || "error" in businessStructures) return serverError;
+  // TODO: define schema of data returned from endpoint
+  const userOperatorData: any | { error: string } =
+    await getUserOperatorFormData(params.id);
+
+  if (
+    "error" in userData ||
+    "error" in businessStructures ||
+    "error" in userOperatorData
+  )
+    return;
 
   const businessStructuresList = businessStructures?.map(
     (businessStructure: BusinessStructure) => ({
       id: businessStructure.name,
       label: businessStructure.name,
     }),
   );
-
+  // FIXME: this data is bogus. Replace with genuine data.
+  userOperatorData.is_senior_officer = "true";
+  userOperatorData.operator_has_parent_company = "no";
   // If operator has an admin, use the single page form to show the user information
-  return params?.id ? (
-    <UserOperatorForm schema={userOperatorPage2} formData={userData} />
-  ) : (
+  // TODO: is there a better/smarter way to distinguish between what form component should be displayed?
```

Comment:
does this TODO comment still hold?

Reply:
I believe this comment has been addressed and removed but not certain @marcellmueller

Reply:
This comment has been removed as the logic to choose which form is displayed is in `UserOperatorMultistepForm` as we needed to do it in a client component so we could use `useSession`.